modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php

```diff
@@ -125,13 +125,20 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
       // Link.
       $link_render_array = [];
       $link_url = '';
+      $current_path = \Drupal::service('path.current')->getPath();
       if ($item->link_title || $item->link_uri) {
         if (str_starts_with($item->link_uri ?? '', '/' . PublicStream::basePath())) {
           // Link to public file: use fromUri() to get the URL.
           $link_url = Url::fromUri(urldecode('base:' . $item->link_uri));
         }
         else {
-          $link_url = $this->pathValidator->getUrlIfValid($item->link_uri ?? '<none>');
+          // Check if the link is an anchor within the current page.
+          if (str_starts_with($item->link_uri, "#")) {
+            $link_url = $this->pathValidator->getUrlIfValid($current_path . $item->link_uri);
```

Comment:
To prevent an error from occurring when `$item->link_uri` is null, we can add a null coalescing operator to line 136 (just like the previous if statement above). In line 137, we can simplify the variable assignment and use the Drupal [Url::fromUserInput](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Url.php/function/Url%3A%3AfromUserInput/10) function, which was designed for this use case. Then we won't need the new `$current_path` variable.

```suggestion
          if (str_starts_with($item->link_uri ?? '', "#")) {
            $link_url = Url::fromUserInput($item->link_uri);
```